src/main/java/io/github/zap/commons/event/BukkitProxy.java

```diff
@@ -0,0 +1,116 @@
+package io.github.zap.commons.event;
+
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.plugin.EventExecutor;
+import org.bukkit.plugin.Plugin;
+import org.bukkit.plugin.RegisteredListener;
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Objects;
+import java.util.logging.Level;
+
+class BukkitProxy<T extends org.bukkit.event.Event> extends SimpleEvent<T> {
+    private final Listener listener = new Listener() {};
+
+    private final Plugin plugin;
+    private final Class<T> bukkitEventClass;
+    private final EventPriority priority;
+    private final boolean ignoreCancelled;
+
+    private boolean eventRegistered = false;
+    private boolean reflectionFailed = false;
+    private HandlerList handlerList;
+    private RegisteredListener registeredListener;
+
+    BukkitProxy(@NotNull Plugin plugin, @NotNull Class<T> bukkitEventClass, @NotNull EventPriority priority,
+                boolean ignoreCancelled) {
```

Comment:
This might be unnecessary but it is more flexible when you can pass canceled events as an event argument (similar to `XEvent#isCanceled()`) where handlers can decide if they want to act upon it

Reply:
i might be misunderstanding what you're saying - but bukkit events do support cancellation and proxyevents should work fine with that

Reply:
Yeah maybe my point is unclear, but I am talking about the EventHandler (proxied of Listener) ability to be invoked whether the event has been canceled or not from reading the javadocs I suspect that passing false here will prevent canceled event will prevent Listener to be invoked

![image](https://user-images.githubusercontent.com/33594017/134332354-6c4ad36e-8e1f-4801-8d10-caef24e12718.png)

Reply:
yeah that's basically how it works i think

Reply:
In short I think it is better to always pass `false` in line 57

Reply:
i remember running into problems with 'mysteriously' not receiving events (because mythicmobs cancelled them) so we almost always have to ignore cancellation, at least for zombies

but this is meant to also be a public API, and it's probably better to respect bukkit default